src/libraries/System.Text.Json/gen/Model/TypeRef.cs

```diff
@@ -0,0 +1,50 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Text.Json.Reflection;
+using Microsoft.CodeAnalysis;
+
+namespace System.Text.Json.SourceGeneration
+{
+    /// <summary>
+    /// An equatable value representing type identity.
+    /// </summary>
+    [DebuggerDisplay("Name={Name}")]
+    public sealed class TypeRef : IEquatable<TypeRef>
+    {
+        public TypeRef(Type type)
+        {
+            Debug.Assert(type is TypeWrapper);
+            var wrapper = ((TypeWrapper)type);
+
+            Name = type.Name;
+            FullyQualifiedName = type.GetCompilableName();
+            IsValueType = wrapper.IsValueType;
+            TypeKind = wrapper.Symbol.TypeKind;
+            SpecialType = wrapper.Symbol.SpecialType;
+            CanContainNullableReferenceAnnotations = type.CanContainNullableReferenceTypeAnnotations();
```

Comment:
Why not a record for this type? Seems like pointwise equality still holds given these members.

Reply:
It would be inefficient because comparing the fully qualified name should suffice for types.